### ISSUE TITLE
Fix stock vinyl related issues

### ISSUE THREAD

**Issue by LeoCodes21** [SoapboxRaceWorld/soapbox-race-core#20]
Some presets include "stock" vinyls that are not available in the shop (usually, at least). That by itself is not an issue. However, there appears to be an internal inconsistency in data, since attempting to perform any customization will usually result in a "Invalid Basket" error.

**Comment by LeoCodes21** [comment]
Cannot reproduce locally, even after re-introducing the "invalid basket" code. More testing is necessary.

**Comment by LeoCodes21** [comment]
Closing this for now, further testing is pending.

**Comment by driftyz700** [comment]
when i am trying to restore the mr2 gymkhana's stock vinyls SPECIFICALLY using vinylmanager.exe game crashes when hitting confirm, i believe that is kinda relevant to this case.

**Comment by LeoCodes21** [comment]
The game crashes because vinyl manager is unable to locate one of the vinyls. Most servers do not have the stock vinyls in the shop.

**Comment by LeoCodes21** [comment]
This issue has been resolved. A technical explanation: 

When a car is purchased, the game assumes that the server is in sync with the client regarding presets, and that the server is able to handle customization. Normally that assumption is true. However, presets with full-body vinyls present an interesting difference: although the raw data indicates a scale of (0, 0) **\[1\]**, the game client makes a change:
(pseudocode for reader understanding)
```cpp
  if ( this->ScaleX == 0 )
    this->ScaleX = 3277;
  if ( this->ScaleY == 0 )
    this->ScaleY = 3277;
```

When customization requests were submitted to the server, the server didn't recognize the meaning of 3277, and assumed the vinyl was _added_. This caused the server to reject any attempt at customization, because it didn't recognize the apparently-added vinyl.

This has been resolved with 2 steps:
1. Updating the server so that it [correctly handles 3277 scale](https://github.com/SoapboxRaceWorld/soapbox-race-core/commit/214fc221c98c55fd74de70a100a7bfff4ce81814).
2. Updating server-side basket definitions to be in sync with decoded client presets (there was another issue that I encountered during testing; this second step has resolved that issue)

This was certainly very interesting to debug and resolve. Updated basket data will be published at some point.

**\[1\]**: (x, y); ScaleX, ScaleY
